utoipa-gen/src/schema/component.rs

```diff
@@ -796,16 +796,21 @@ where
                             .map(|attributes| attributes.is_inline())
                             .unwrap_or(false);
 
-                        if is_inline {
-                            let component_name_ident = &self.component_part.ident;
-                            tokens.extend(quote! {
-                                <#component_name_ident as utoipa::Component>::component()
-                            });
+                        // When users wishes to hinder the actual type with Any type render a generic `object`
+                        if component_part.is_any() {
+                            tokens.extend(quote! { utoipa::openapi::ObjectBuilder::new() })
                         } else {
-                            let name = &*self.component_part.ident.to_string();
-                            tokens.extend(quote! {
-                                utoipa::openapi::Ref::from_component_name(#name)
-                            })
+                            if is_inline {
+                                let component_name_ident = &self.component_part.ident;
```

Comment:
remove the `&self` so it will work with the `override type` as well which is defined beginning of the function :)
```suggestion
                                let component_name_ident = component_part.ident;
```